autoclass/autodict_.py

```diff
@@ -257,9 +257,6 @@ def execute_autodict_on_class(cls,                       # type: Type[T]
                     '__contains__',
                     'get',
                     'items',
```

Comment:
This removal makes the python 2 test fail since this if/else statement is specifically for python 2. You can probably safely put it back

Reply:
Thanks for pointing me in the right direction. 
Under Python3.9, the code: `cls.__bases__ = new_bases` throws `TypeError: __bases__ assignment: 'Mapping' deallocator differs from 'object'` which I'm guessing might be the exception you're trying to catch.

The `cls.__bases__ = with_metaclass(type(cls), *new_bases)` throws `TypeError: can only assign tuple to Experiment.__bases__, not metaclass` (`Experiment` is the name of my class where I'm using autoclass)

I naively tried to make it a tuple as `cls.__bases__ = (with_metaclass(type(cls), *new_bases),)` and I get the exception `TypeError: __bases__ assignment: 'temporary_class' deallocator differs from 'object'`

I don't know what to try next, but maybe this gives you some information?

Reply:
Thanks a lot for providing this ! Well this does not look good. I understand why you ended up fixing the last part of the try/except.

Not sure that I can fix this very soon. I'll have a look in the upcoming weeks. 

If you wish to have a stab at it in between, you can try to 

 - revert the changes you did,  
 - create a boolean version flag (`PY38 = sys.version_info >= (3, 8)`) 
 - use that flag in the last "except" (just where you did your first mod) to skip the method names that are not relevant anymore, and to not use `.im_func`.

Also note that a more recent lib that I consider better than `autoclass` for my personal use is https://smarie.github.io/python-pyfields/ . It also provides an [`@autoclass` decorator](https://smarie.github.io/python-pyfields/#b-autoclass). Instead of fiddling with the class bases, this decorator adds  `to_dict` and `from_dict` methods.

Reply:
Hi Sylvain, I've moved to `pyfields` as you suggested, that solved the issue for me. I will not grapple with trying to address the issue in `autoclass`. Feel free to close the PR if you want.